resources/run_sampling.rb

```diff
@@ -126,30 +128,19 @@ def perform_sampling(params, num_samples, tsvfiles, project_dir_name)
     return results_data
   end
 
-  def get_bldgs_by_dependency_values(results_data, dep_hashes, num_samples, results_data_cols)
+  def get_bldgs_by_dependency_values(results_data, dep_hashes, results_data_cols)
     # Returns a hash with key:dep_hash, value:Array[bldgs]
 
-    # Initialize
-    bldgs_hash = {}
-    dep_hashes.each do |dep_hash|
-      bldgs_hash[dep_hash] = []
+    data = []
+    dep_hashes[0].each do |dep_name, dep_val|
+      data << results_data[results_data_cols[dep_name]]
     end
+    data = data.transpose
 
-    # For each building, assign to appropriate dep_hash
-    for bldg_num in 1..num_samples
-      dep_hashes.each do |dep_hash|
-        match = true
-        dep_hash.each do |dep_name, dep_value|
-          next if results_data[results_data_cols[dep_name]][bldg_num] == dep_value
-
-          match = false
-          break
-        end
-        if match
-          bldgs_hash[dep_hash] << bldg_num
-          break
-        end
-      end
+    bldgs_hash = {}
+    data[1..-1].each_with_index do |bldg, idx|
+      bldgs_hash[bldg] = [] if bldgs_hash[bldg].nil?
+      bldgs_hash[bldg] << idx + 1
     end
```

Comment:
This is the real performance improvement for sampling. Previously we had three nested loops; by transposing the data up front, it can be reduced to a single loop.